docs/software-development-guide.md

```diff
@@ -0,0 +1,189 @@
+
+---
+
+## I. Guiding Principles
+
+### PRs are communication artifacts
+A PR should always make its **intent, scope, and status** explicit.
+The diff alone is not sufficient context.
+
+### Uncertainty should be surfaced early
+Open questions, design concerns, or architectural uncertainty should appear early in the PR—not discovered late during review.
+
+### Splitting and closing PRs is normal
+Large, blocked, or outdated PRs should be split or closed without stigma.
+Closing a PR is a decision, not a failure.
+
+### Momentum is a shared responsibility
+Authors and reviewers are jointly responsible for moving PRs toward a clear outcome.
+
+### Explicit decisions over silent waiting
+A stalled PR should always converge toward a visible decision: proceed, split, block, or close.
+
+---
+
+## II. Common Challenges
+
+### Stale or silent PRs
+PRs remain open without activity, status, or clear ownership.
+
+### PRs with unclear or growing scope
+PRs expand during review or surface design uncertainty late, making review slow and difficult.
+
+### Blocked PRs without visible blockers
+Dependencies on other work, people, or decisions are not made explicit.
+
+### Loss of context across PRs and repositories
+PRs, issues, and follow-ups are not consistently linked.
+
+### Limited visibility into parallel work
+Team members are unaware of related or overlapping work.
+
+### Reviews that do not advance toward a decision
+Feedback acknowledges work without clearly approving, blocking, or requesting concrete changes.
+
+### Reluctance to close or rework PRs
+PRs remain open even when direction changes or scope becomes unclear.
+
+---
+
+## III. Branching, Conflict Resolution & PR Maintenance
+
+### Branching Model
+
+- Development primarily happens on **short-lived branches** created from `main`.
+- Branching off a working branch is acceptable during the draft phase (e.g. to try alternatives or propose fixes).
+- Branch names should use a short, hyphen-separated description.
+- Each branch should address **one logical change only**.
+
+---
+
+### Conflict Resolution: Rebasing and Merging
+
+Both `git rebase` and `git merge` are valid tools. The choice should be **pragmatic**.
+
+**Default guidance**
+- Rebasing onto `main` is the default recommendation, especially for newer contributors.
+- Merging `main` into a feature branch is also acceptable and sometimes preferable.
+
+**Rebasing is often easier when**
+- PRs are small or medium-sized
+- Conflicts are limited
+- A clean, reviewable diff matters
+
+**Merging may be preferable when**
+- Branches are long-lived
+- Conflicts are complex or repeatedly reoccur
+- Preserving the development sequence aids understanding
+- The branch already integrates many upstream changes
+
+Because PRs are merged via **Squash & Merge**, neither approach affects the final history of `main`.
+
+**Key expectation**
+A feature branch should include **all relevant changes from the target branch**, not merely be conflict-free.
```

Comment:
When it's conflict-free, it means that the `git` algo can handle the rest. I'm reading this as "don't destroy any existing infrastructure, unless this was the express purpose of the PR".

Reply:
I tried to clarify the intent now